api/v1beta1/cluster_types.go

```diff
@@ -193,9 +193,15 @@ type MachineDeploymentVariables struct {
 type ClusterNetwork struct {
 	// APIServerPort specifies the port the API Server should bind to.
 	// Defaults to 6443.
+	// Deprecated: use LocalAPIServerPort instead
 	// +optional
 	APIServerPort *int32 `json:"apiServerPort,omitempty"`
 
+	// LocalAPIServerPort specifies the port the local API Server should bind to.
```

Comment:
Similarly, I would add a comment on KubeadmConfig where this value gets used as a default for control plane machine

Reply:
+ to adding a note to clarify `APIServerPort` should not be used as controlplane endpoint, as this is the current CAPA behaviour.

It would be nice to clarify what field to use instead so that we can start supporting the new field in CAPA.

Is it going to be controlPlaneEndpoint.port?

cc @Ankitasw